Doc/whatsnew/3.11.rst

```diff
@@ -493,6 +494,213 @@ Optimizations
+
+In Python 3.11, the core modules essential for Python startup are "frozen".
+This means that their code objects (and bytecode) are statically allocated
+by the interpreter. This reduces the steps in module execution process to this:
+
+.. code-block:: text
+
+   Statically allocated code object -> Evaluate
+
+Interpreter startup is now 10-15% faster in Python 3.11. This positively
+impacts short-running programs using Python. Such as ``python -m venv ...``, or
+``python -m pip ...``.
+
+(Contributed by Eric Snow, Guido van Rossum and Kumar Aditya in numerous issues.)
+
+
+Faster Runtime
+--------------
+
+Cheaper, lazy Python frames
+~~~~~~~~~~~~~~~~~~~~~~~~~~~
+Python frames are created whenever Python calls a Python function. This frame
+holds execution information. The following are new frame optimizations:
+
+- Streamlined the frame creation process.
+- Avoided memory allocation by generously re-using frame space on the C stack.
+- Streamlined the internal frame object to only contain essential information.
+  Frames previously held extra debugging and memory management information.
+  Old-style frames are now created only when required by debuggers. For most
+  user code, no frames are created at all.
+
+Nearly all Python functions calls have sped up significantly.
+This resulted in a 3-7% improvement in pyperformance.
+(Contributed by Mark Shannon in :issue:`44590`.)
+
+.. _inline-calls:
+
+Inlined Python function calls
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+During a Python function call, Python will call a C function to interpret that
+function's code, similar to how a user would call ``eval`` to run arbitrary
+code.
+
+In 3.11, when Python detects code calling another Python function,
+it sets up a new frame, and "jumps" to the new code inside the new frame. This
+avoids calling the C interpreting function altogether.
+
+Python function calls now consume almost no C stack space. This speeds up Python
+to Python function calls. In simple recursive functions like fibonacci or
+factorial, a 1.7x speedup was observed. This also means recursive functions
+can recurse significantly deeper, assuming the recursion limit and memory limit
+is not exceeded. This resulted in a 1-3% improvement in pyperformance.
+(Contributed by Pablo Galindo and Mark Shannon in :issue:`45256`.)
+
+PEP 659: Specializing Adaptive Interpreter
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+:pep:`659` is one of the key parts of the faster CPython project. The general
+idea is that while Python is a dynamic language, most code have regions where
+objects and types rarely change. This concept is known as *type stability*.
+
+At runtime, Python will try to look for common patterns and type stability
+in the executing code. Python will then replace the current operation with a
+more specialized one. This specialized operation use fast paths available only
+to those use cases/types and will generally outperform their generic
```

Comment:
```suggestion
more specialized one. This specialized operation uses fast paths available only
to those use cases/types, which will generally outperform their generic
```
Also maybe remove "generally"

Reply:
Thanks for fixing those embarrassing grammar mistakes.

I say "generally" because it's true :). There are some cases where specialized opcodes don't speed up or are slower (e.g. generic vectorcalled C functions, or polymorphic code).